<?xml version="1.0" ?>

<launch>
  <arg name="for_real" default="true"/>
  <arg name="gripper_present" default="true" />
  <arg name="netft_present" default="true" />
  <arg name="publish_rate" default="30" />
  <arg name="controller_executable" default="pdcontroller_gripper_netft" />
  
  <group ns="panda1">
      <rosparam command="load" file="$(find controller_panda_mti)/config/$(env PANDA_MAC).yaml" />
      <!-- Override the yaml parameters so that controllers different to the default can be tested:  -->
      <param name="withGripper" value="$(arg gripper_present)"/>
      <param name="withNetft" value="$(arg netft_present)"/>

      <param if="$(arg gripper_present)"     name="robot_description" command="$(find xacro)/xacro  $(find panda_dynamics_model)/urdf/panda_arm_hand_alt_dyn.urdf.xacro" />
      <param unless="$(arg gripper_present)" name="robot_description" command="$(find xacro)/xacro  $(find panda_dynamics_model)/urdf/panda_arm_alt_dyn.urdf.xacro"/>
<!--      <param if="$(arg gripper_present)"     name="robot_description" command="$(find xacro)/xacro  $(find franka_description)/robots/panda_arm_hand.urdf.xacro" />-->
<!--      <param unless="$(arg gripper_present)" name="robot_description" command="$(find xacro)/xacro  $(find franka_description)/robots/panda_arm.urdf.xacro"/>-->



      <node     if="$(arg for_real)"   name="pdcontroller" pkg="controller_panda_mti"   launch-prefix="chrt -f 40"  type="$(arg controller_executable)" required="true"  output="screen" />
      <node unless="$(arg for_real)" name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher">
        <param  if="$(arg load_gripper)" name="publish_default_positions" value="true" />
      </node>
 </group>
 
</launch>
